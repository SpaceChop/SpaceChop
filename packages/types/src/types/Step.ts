import * as t from 'runtypes';
import CompressConfig from './operations/compress';
import CropConfig from './operations/crop';
import FillConfig from './operations/fill';
import FitConfig from './operations/fit';
import FormatConfig from './operations/format';
import ResizeConfig from './operations/resize';
import StripConfig from './operations/strip';

const Step = t.Partial({
  $compress: CompressConfig,
  $crop: CropConfig,
  $fill: FillConfig,
  $fit: FitConfig,
  $format: FormatConfig,
  $resize: ResizeConfig,
  $strip: StripConfig,
}).withConstraint(
  (n) => {
    if (Object.keys(n).length > 1) {
      return 'please only one operation per step';
    }
    if (Object.keys(n).length === 0) {
      return 'please no empty step';
    }
    return true;
  },
);

export type Step = t.Static<typeof Step>;
export default Step;
