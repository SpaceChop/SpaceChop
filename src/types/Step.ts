import * as t from 'runtypes';
import CompressConfig from '../operations/compress/types';
import CropConfig from '../operations/crop/types';
import FillConfig from '../operations/fill/types';
import FitConfig from '../operations/fit/types';
import FormatConfig from '../operations/format/types';
import ResizeConfig from '../operations/resize/types';
import StripConfig from '../operations/strip/types';

const Step = t.Partial({
  $compress: CompressConfig,
  $crop: CropConfig,
  $fill: FillConfig,
  $fit: FitConfig,
  $format: FormatConfig,
  $resize: ResizeConfig,
  $strip: StripConfig,
}).withConstraint(
  (n) => {
    if (Object.keys(n).length > 1) {
      return 'please only one operation per step';
    }
    if (Object.keys(n).length === 0) {
      return 'please no empty step';
    }
    return true;
  },
);

export type Step = t.Static<typeof Step>;
export default Step;
