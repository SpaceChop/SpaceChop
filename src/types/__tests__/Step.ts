import Step from '../Step';

describe('validate Step', () => {
  describe('invalid', () => {
    it('no double operations in step', () => {
      const config = {
        $crop: {
          width: 100,
          height: 100,
        },
        $resize: {
          width: 100,
          height: 100,
        },
      };
      expect(() => Step.check(config)).toThrowErrorMatchingSnapshot();
    });
  });
});
