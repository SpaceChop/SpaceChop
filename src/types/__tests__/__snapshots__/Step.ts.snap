// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Step invalid no double operations in step 1`] = `"please only one operation per step"`;
